src/avram/database.cr

```diff
@@ -180,7 +189,7 @@ abstract class Avram::Database
 
   # :nodoc:
   def transaction : Bool
-    if current_transaction
+    if current_transaction.try(&.joinable?)
```

Comment:
transaction is "joinable" if it does not require a sub-transaction to be made. We have always considered nested transactions to be "joinable" though it wasn't encoded. The new thing is specifying that a transaction is not joinable and a sub-transaction should be made.

This was added so that specs verifying failed operations and other transaction rollback scenarios would still work. The "root" transaction that is added in the spec helper is not joinable so the transaction made in an operation is forced to be sub